pipeline_tools/pipelines/smartseq2/smartseq2.py

```diff
@@ -85,3 +85,87 @@ def _get_content_for_ss2_input_tsv(bundle_uuid, bundle_version, dss_url, http_re
     sample_id = metadata_utils.get_sample_id(primary_bundle)
     fastq_1_url, fastq_2_url = get_urls_to_files_for_ss2(primary_bundle)
     return fastq_1_url, fastq_2_url, sample_id
+
+
+def create_ss2_se_input_tsv(
+    bundle_uuid, bundle_version, dss_url, input_tsv_name='inputs.tsv'
+):
+    """Create TSV of Smart-seq2 inputs.
+
+    Args:
+        bundle_uuid (str): The bundle uuid
+        bundle_version (str): The bundle version
+        dss_url (str): The url for the DCP Data Storage Service
+        input_tsv_name (str): The file name of the input TSV file. By default, it's set to 'inputs.tsv',
+                              which will be consumed by the pipelines.
+
+    Returns:
+        None: this function will write the TSV file of cloud paths for the input files.
+
+    Raises:
+        requests.HTTPError: for 4xx errors or 5xx errors beyond the timeout
+    """
+    fastq_url, sample_id = _get_content_for_ss2_se_input_tsv(
+        bundle_uuid, bundle_version, dss_url, HttpRequests()
+    )
+
+    print('Creating input map')
+    with open(input_tsv_name, 'w') as f:
+        f.write('fastq\tsample_id\n')
+        f.write('{0}\t{1}\n'.format(fastq_url, sample_id))
+    print('Wrote input map to disk.')
+
+
+def get_urls_to_files_for_ss2_se(bundle):
+    """Return the direct urls to the input fastq files for SmartSeq2 pipeline.
+
+    Args:
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
+
+    Returns:
+         tuple: A tuple consisting of the url to the input fastq file
+         respectively.
+    """
+    fastq_url = None
+    sequence_files = bundle.sequencing_output
+
+    for sf in sequence_files:
```

Comment:
You can remove the check for `if fastq_url`, and move the return statement under `if sf.read_index == 'read1'` to simplify this a bit:
```
if sf.read_index == 'read1':
    return sf.manifest_entry.url
```

Reply:
Thanks, updated